File: src/test/java/tech/macil/minecraft/rcon/util/NLRequiringBufferedReaderTest.java
package tech.macil.minecraft.rcon.util;

import com.google.common.collect.Lists;
import org.junit.Test;

import java.io.StringReader;
import java.util.stream.Collectors;

import static org.junit.Assert.*;

public class NLRequiringBufferedReaderTest {
    @Test
    public void readLine() throws Exception {
        NLRequiringBufferedReader r = new NLRequiringBufferedReader(new StringReader(
                "foo\n\nbar x x\nbaz"));
        assertEquals("foo", r.readLine());
        assertEquals("", r.readLine());
        assertEquals("bar x x", r.readLine());
        assertEquals(null, r.readLine());
    }

    @Test
    public void lines() throws Exception {
        NLRequiringBufferedReader r = new NLRequiringBufferedReader(new StringReader(
                "foo\n\nbar x x\nbaz"));
        assertEquals(
                Lists.newArrayList("foo", "", "bar x x"),
                r.lines().collect(Collectors.toList())
        );
    }
}

File: src/test/kotlin/tech/macil/minecraft/rcon/util/NLRequiringBufferedReaderTest.kt
package tech.macil.minecraft.rcon.util

import com.google.common.collect.Lists
import org.junit.Test

import java.io.StringReader
import java.util.stream.Collectors

import org.junit.Assert.*

class NLRequiringBufferedReaderTest {
    @Test
    fun readLine() {
        val r = NLRequiringBufferedReader(StringReader(
                "foo\n\nbar x x\nbaz"))
        assertEquals("foo", r.readLine())
        assertEquals("", r.readLine())
        assertEquals("bar x x", r.readLine())
        assertNull(r.readLine())
    }

    @Test
    fun lines() {
        val r = NLRequiringBufferedReader(StringReader(
                "foo\n\nbar x x\nbaz"))
        assertEquals(
                Lists.newArrayList("foo", "", "bar x x"),
                r.lines().collect(Collectors.toList())
        )
    }
}